tools/src/commands/SyncBundledNativeModules.ts

```diff
@@ -0,0 +1,239 @@
+    },
+  ]);
+  if (!confirmed) {
+    logger.info('No worries, come back soon!');
+    process.exit(1);
+  }
+}
+
+async function resolveSecretAsync(): Promise<string> {
+  if (process.env.EXPO_SDK_NATIVE_MODULES_SECRET) {
+    return process.env.EXPO_SDK_NATIVE_MODULES_SECRET;
+  }
+
+  logger.info(
+    `We need the secret to authenticate you with Expo servers.\nPlease set the ${chalk.green(
+      'EXPO_SDK_NATIVE_MODULES_SECRET'
+    )} env var if you want to skip the prompt in the future.`
+  );
+
+  const { secret } = await inquirer.prompt<{ secret: string }>([
+    {
+      type: 'password',
+      name: 'secret',
+      message: `Secret:`,
+      validate: (val) => (val ? true : 'The secret cannot be empty'),
+    },
+  ]);
+  return secret;
+}
+
+async function resolveTargetSdkVersionAsync(): Promise<string> {
+  const expoPackageJsonPath = path.join(EXPO_PACKAGE_PATH, 'package.json');
+  const contents = await JsonFile.readAsync<Record<string, string>>(expoPackageJsonPath);
+  const majorVersion = semver.major(contents.version);
+
+  const sdkVersion = `${majorVersion}.0.0`;
+
+  const { confirmed } = await inquirer.prompt<{ confirmed: boolean }>([
+    {
+      type: 'confirm',
+      name: 'confirmed',
+      message: `Do you want to sync bundledNativeModules.json for ${chalk.green(
+        `SDK ${sdkVersion}`
+      )}?`,
+      default: true,
+    },
+  ]);
+
+  if (!confirmed) {
+    logger.info('No worries, come back soon!');
+    process.exit(1);
+  } else {
+    return sdkVersion;
+  }
+}
+
+async function readBundledNativeModulesAsync(): Promise<BundledNativeModules> {
+  const bundledNativeModulesPath = path.join(EXPO_PACKAGE_PATH, 'bundledNativeModules.json');
+  return await JsonFile.readAsync<BundledNativeModules>(bundledNativeModulesPath);
+}
+
+async function getCurrentBundledNativeModules(
+  env: Env,
+  sdkVersion: string
+): Promise<BundledNativeModulesList> {
+  const baseApiUrl = resolveBaseApiUrl(env);
+  const result = await fetch(`${baseApiUrl}/--/api/v2/sdks/${sdkVersion}/native-modules`);
+  const resultJson: GetBundledNativeModulesResult = await result.json();
+  return resultJson.data;
+}
+
+async function compareAndConfirmAsync(
+  current: BundledNativeModulesList,
+  next: BundledNativeModulesList
+): Promise<void> {
+  const currentMap = current.reduce((acc, i) => {
+    acc[i.npmPackage] = i;
+    return acc;
+  }, {} as Record<string, NativeModule>);
+
+  logger.info('Changes:');
+  let hasChanges = false;
+  for (const { npmPackage, versionRange } of next) {
+    if (!(npmPackage in currentMap)) {
+      hasChanges = true;
+      logger.info(` - ${npmPackage}: ${chalk.red(`(none)`)} -> ${chalk.green(versionRange)}`);
+    } else if (versionRange !== currentMap[npmPackage].versionRange) {
+      hasChanges = true;
+      logger.info(
+        ` - ${npmPackage}: ${chalk.red(currentMap[npmPackage].versionRange)} -> ${chalk.green(
+          versionRange
+        )}`
+      );
+    }
```

Comment:
```suggestion
    // currentMap might either hold a different version of the package or have no entry for given package
    if (versionRange !== currentMap[npmPackage]?.versionRange) {
      hasChanges = true;
      logger.info(` - ${npmPackage}: ${chalk.red(currentMap[npmPackage]?.versionRange ?? `(none)`)} -> ${chalk.green(versionRange)}`);
    }
```
Why to duplicate the very same code block? 🤔